File: API/Dtos/AddClinicDto.cs
using API.Entities;

namespace API.Dtos
{
    public class AddClinicDto
    {
        public string? Name { get; set; }
        public string? Address { get; set; }
        public string? NIP { get; set; }
        public bool? IsNFZ { get; set; }
        public bool? IsPrivate { get; set; }
        public ICollection<ClinicDoctor>? ClinicDoctors { get; set; }
        public DateTime? OfficeHoursStartDate { get; set; }
        public DateTime? OfficeHoursEndDate { get; set; }
        public List<int>? WorkingDays { get; set; }
        public string? PhoneNumber { get; set; }
    }
}

File: client/medplanner-app/src/app/DTOs/add-clinic-dto.ts
import { ClinicDoctor } from "../models/clinic-doctor";

export interface AddClinicDto {
    id?: string;
    name?: string;
    address?: string;
    nip?: string;
    isNFZ?: boolean;
    isPrivate?: boolean;
    clinicDoctors?: ClinicDoctor[];
    officeHoursStartDate?: Date;
    officeHoursEndDate?: Date;
    workingDays?: DayOfWeek[];
    phoneNumber?: string;
}


export enum DayOfWeek {
    Sunday = 0,
    Monday = 1,
    Tuesday = 2,
    Wednesday = 3,
    Thursday = 4,
    Friday = 5,
    Saturday = 6
}
